File: python-backend/app/models/data_metrics_config.py
"""DataMetricsConfig ORM model."""

from sqlalchemy import BigInteger, Integer, String
from sqlalchemy.orm import Mapped, mapped_column

from app.core.database import Base


class DataMetricsConfig(Base):
    __tablename__ = "data_metrics_config"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    thresholds: Mapped[str] = mapped_column(String, nullable=False)
    overtime_hour: Mapped[int] = mapped_column("overtimeHour", Integer, nullable=False)
    labels: Mapped[str] = mapped_column(String, nullable=False)
    colors: Mapped[str] = mapped_column(String, nullable=False)
    created_at: Mapped[int] = mapped_column(BigInteger, nullable=False)
    updated_at: Mapped[int] = mapped_column(BigInteger, nullable=False)

File: golang-backend/internal/model/data_metrics_config.go
package model

type DataMetricsConfig struct {
	ID           int64  `gorm:"primaryKey;autoIncrement" json:"id"`
	Thresholds   string `gorm:"not null" json:"thresholds"`
	OvertimeHour int    `gorm:"not null" json:"overtimeHour"`
	Labels       string `gorm:"not null" json:"labels"`
	Colors       string `gorm:"not null" json:"colors"`
	CreatedAt    int64  `gorm:"column:created_at" json:"createdAt"`
	UpdatedAt    int64  `gorm:"column:updated_at" json:"updatedAt"`
}

func (DataMetricsConfig) TableName() string { return "data_metrics_config" }
